akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/Behaviors.scala

```diff
@@ -227,8 +227,9 @@ object Behaviors {
    *
    * See also [[akka.actor.typed.Logger.withMdc]]
    */
-  def withMdc[T](mdcForMessage: T ⇒ Map[String, Any])(behavior: Behavior[T]): Behavior[T] =
-    withMdc[T](Map.empty[String, Any], mdcForMessage)(behavior)
+  // note: cannot be multi-param-list because it breaks type inference
+  def withMdcPerMessage[T](mdcForMessage: T ⇒ Map[String, Any], behavior: Behavior[T]): Behavior[T] =
```

Comment:
As soon as we have the `T => something` we need to put the `Behavior[T]` in the same parameter list or else type inference fails and we have to put explicit types somewhere.